### PR TITLE
Handle Android camera permissions error code

### PR DESCRIPTION
**Pull request by camsim99**
Handles the new camera permissions code for Android that represents an ongoing request that was introduced in PR [#5640](https://github.com/flutter/plugins/pull/5640). 

Part of Issue [#101384](https://github.com/flutter/flutter/issues/101384). 

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [ ] I signed the [CLA].
- [ ] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [ ] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/main/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by camsim99** [comment]
@stuartmorgan Reading through the wiki, it seems like this PR should be reviewed first? Then the changes with `camera_android` and `camera_web`, and then the version bumps of `camera_android` and `camera_web` with the `camera` changes to follow. Is that correct?

**Comment by bparrishMines** [comment]
@camsim99 This still needs a test, right? I can review it once it has one.

File: packages/camera/camera_platform_interface/test/method_channel/method_channel_camera_test.dart [hunk 1]
```diff
@@ -56,37 +56,31 @@ void main() {
       });
 
       test(
-          'Should throw CameraException when create throws a PlatformException',
+          'Should throw StateError when create is called while camera permissions request ongoing',
```
Comment (camsim99) [inline packages/camera/camera_platform_interface/test/method_channel/method_channel_camera_test.dart:59]: This was a duplicate test (see test right below), so I just modified this one.

**Comment by stuartmorgan** [comment]
> @stuartmorgan Reading through the wiki, it seems like this PR should be reviewed first? Then the changes with `camera_android` and `camera_web`, and then the version bumps of `camera_android` and `camera_web` with the `camera` changes to follow. Is that correct?

This PR isn't changing any interfaces, so there doesn't need to be any chaining for PRs AFAICT.

I don't think we even need this PR though, since none of our implementations should be using the default method channel implementation now. Do we need to change the behavior of third-party camera implementations? If not, you can just skip this unless I'm missing something.

**Pull request closed without merging**